src/DependencyInjection/Compiler/ServiceAnnotationPass.php

```diff
@@ -35,16 +35,22 @@ public function process(ContainerBuilder $container): void
 
         $this->annotationReader = $container->get('annotation_reader');
 
-        $services = array_keys($container->findTaggedServiceIds('terminal42_service_annotation'));
+        foreach ($container->getDefinitions() as $id => $definition) {
```

Comment:
So now you are iterating over all existing definitions? Not sure if that is correct (or performant).

Reply:
Yes its iterating over all services, but only at compile time. That's the [same as Symfony does](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Compiler/ResolveInstanceofConditionalsPass.php#L38) when handling autoconfiguration. And the reason why we no longer need a tag or a marker interface. I didn't find any noticeable speed difference in my local tests (like it didn't take 5 or 10 seconds).

Reply:
Yeah, performance is no issue here.